### PR TITLE
Add locked line

### PR DESCRIPTION
**Pull request by nishasy**
## Summary:
Add the UI to InteractiveGraphEditor to add/remove/edit locked lines.

Issue: https://khanacademy.atlassian.net/browse/LEMS-1740

## Test plan:
`yarn jest packages/perseus-editor/src/widgets/__tests__/interactive-graph-editor-locked-figures.test.tsx`

Storybook
- http://localhost:6006/?path=/story/perseus-editor-widgets-interactive-graph-editor--with-locked-lines
- http://localhost:6006/?path=/story/perseus-editor-editorpage--demo
  - Click "Add a widget"
  - Click "Interactive graph"
  - Click "interactive-graph 1"
  - Choose "Line Segment(s)" as the type of graph
  - Use "Add locked figure" dropdown to add a line

Experience fully in the storybook preview in the checks below.

image

### CODE REVIEW

File: packages/perseus-editor/src/widgets/__tests__/interactive-graph-editor.test.tsx
Comment (nishasy) [inline packages/perseus-editor/src/widgets/__tests__/interactive-graph-editor.test.tsx:1]: This file was getting way too long, so I moved the locked figure tests from this file over to interactive-graph-editor-locked-figures.test.tsx

File: packages/perseus-editor/src/components/locked-line-settings.tsx
Comment (mark-fitzgerald) [inline packages/perseus-editor/src/components/locked-line-settings.tsx:1]: Not sure if this question pertains to this PR or with locked points in particular. When I try to set the start point of the  first locked line (storybook "with locked lines"), I am unable to use decimals in the point location. For instance, I can't set the start point to (4, 1.5), even though the Snap Steps are set to 0.5.

**Review by mark-fitzgerald — APPROVED** [review]
Just one question regarding point coordinates limitation.

File: packages/perseus-editor/src/components/locked-line-settings.tsx
Comment (nishasy) [reply packages/perseus-editor/src/components/locked-line-settings.tsx:1]: @mark-fitzgerald GREAT CATCH, thank you! Looking into it now

**Comment by nishasy** [comment]
The parent pull-request ([#1182](https://github.com/Khan/perseus/pull/1182)) has been merged into `main`, but this branch (`add-locked-line`) now has conflicts with the new base branch. These conflicts must be resolved before checks can complete on this pull-request.

**Pull request merged**